# DAY 2 — INPUTS + DATA TYPES + SIMPLE CALCULATOR
How computers receive information from humans.

#INPUT

Yesterday: computer only showed things.
Today: computer will TALK with user.

Without input Computer only speaks.
With input Human and computer both communicate.

That's how apps,websites and software work.

In [1]:
name = input("Enter your name: ") #input() means "Ask user for information." & 
                                 #Enter your name:"This is the message shown to user."

print(name)                      #name= Means store user answer inside variable called name.

Enter your name: James
James


For Example:
    Imagine food delivery app asks: "What is your address?"

    You type answer and app stores it. 'That is INPUT'

In [2]:
city = input("Enter your city: ")

print(city)

Enter your city: Mumbai
Mumbai


#DATA TYPES

Computers treat words, numbers, true/false differently.
These categories are called 'DATA TYPES'

        MAIN TYPES FOR NOW

| Type    | Meaning        | Example |
| ------- | -------------- | ------- |
| string  | text           | "James" |
| integer | whole number   | 24      |
| float   | decimal number | 5.7     |


STRING (TEXT)

In [3]:
name = "James" #Text uses quotes.

INTEGER (WHOLE NUMBER)

In [4]:
age = 22 #No quotes.

FLOAT (DECIMAL)

In [5]:
height = 5.9 #Decimal numbers.

DIFFERENCE

In [6]:
#This is NUMBER:
age = 22

#This is TEXT:
age = "22"

#Looks same to humans not same to computer. Because math only works properly with numbers.

#SIMPLE CALCULATOR

combine everything

In [7]:
n1 = input("Enter first number: ")

n2 = input("Enter second number: ")

print(n1 + n2)

Enter first number: 10
Enter second number: 5
105


-- What happens? --

Output becomes 105 NOT 15.

Because input() gives TEXT by default.
Computer thinks "10+ "5"

Text joining is called 'concatenation'.

Like  "Hello" + "World"

becomes: HelloWorld

-- HOW TO FIX THIS? --

We convert text into number.

Using ' int() ' means convert into integer number.

In [9]:
n1 = int(input("Enter first number: "))

n2 = int(input("Enter second number: "))

print(n1 + n2)

Enter first number: 10
Enter second number: 5
15


This int() is like 'Computer treat this as number'.

For example:
When we fill online forms age, salary and marks systems often convert text into numbers internally.

Build Simple Calculator

In [10]:
n1 = int(input("Enter first number: "))

n2 = int(input("Enter second number: "))

print("Addition:", n1 + n2)

print("Subtraction:", n1 - n2)

print("Multiplication:", n1 * n2)

Enter first number: 10
Enter second number: 5
Addition: 15
Subtraction: 5
Multiplication: 50


REMEMBER FOREVER

input() → receives information

print() → shows information

variables → store information

int() → converts text into number

Age Calculator

In [11]:
# Input: birth year and current year

birth_year = int(input("Enter your birth year: "))
current_year = int(input("Enter the current year: "))

# Calculate age
age = current_year - birth_year

print("Your age is:", age)

Enter your birth year: 2002
Enter the current year: 2026
Your age is: 24


 Marks Addition Program

In [12]:
# add marks of 3 subjects

m1 = int(input("Enter marks for Subject 1: "))
m2 = int(input("Enter marks for Subject 2: "))
m3 = int(input("Enter marks for Subject 3: "))

total = m1 + m2 + m3
avg= total / 3

print("Total Marks:", total)
print("Average Marks:", avg)


Enter marks for Subject 1: 86
Enter marks for Subject 2: 59
Enter marks for Subject 3: 74
Total Marks: 219
Average Marks: 73.0


Grocery bill calculator

In [13]:
# i = item & p = price & q = quantity

i1_p = float(input("Enter price of item 1: "))
i1_q = int(input("Enter quantity of item 1: "))

i2_p = float(input("Enter price of item 2: "))
i2_q = int(input("Enter quantity of item 2: "))

i3_p = float(input("Enter price of item 3: "))
i3_q = int(input("Enter quantity of item 3: "))

# calculate total for each item
total1 = i1_p * i1_q
total2 = i2_p * i2_q
total3 = i3_p * i3_q

# final bill
grand_total = total1 + total2 + total3

print("Total Bill Amount:", grand_total)

Enter price of item 1: 40
Enter quantity of item 1: 1
Enter price of item 2: 5
Enter quantity of item 2: 3
Enter price of item 3: 10
Enter quantity of item 3: 2
Total Bill Amount: 75.0
